coderd/username/username.go

```diff
@@ -0,0 +1,45 @@
+package username
```

Comment:
I like the idea of extracting validation to a separate package but what sort of precedent do we want to be setting if we have other fields that may be suitable for relocation to a new package? Are we expecting each new field to create its own pkg, to refactor into a common `validate` pkg or do you feel like `username` is sufficiently unique to warrant its own pkg and most everything else should be contained to `coderd`?

Reply:
I feel like this isn't just a case of validation. I extracted this primarily due to the generation that we now need to do with OIDC.

The alternative was exposing `httpapi.ValidateUsername` or something of the sort, which I'm also mostly fine with. Keep in mind we'd have to have a separate function of `httpapi.UsernameFromString` or something like it, but I'm also fine with that. I'll think on it, but if you think that sounds cleaner I'll make it happen!

Reply:
Nah I was just bringing it up to get an idea of how we want to structure packages

Reply:
I actually removed the username package. It felt too much like floaty scope with little benefit. That code is in `httpapi` now, which feels better anyways.

Good comment :+1: